perun-core/src/main/java/cz/metacentrum/perun/core/impl/FacilitiesManagerImpl.java

```diff
@@ -1022,6 +1052,18 @@ public void removeSecurityTeam(PerunSession sess, Facility facility, SecurityTea
 	}
 
 	@Override
+	public void removeService(PerunSession sess, Facility facility, Service service) throws InternalErrorException, ServiceNotExistsException {
+		try {
+			if (0 == jdbc.update("DELETE FROM facility_services WHERE service_id = ? AND facility_id = ?", 
+					service.getId(), facility.getId())) {
+				throw new ServiceNotExistsException(service);
```

Comment:
I believe, that when sql delete returns "0 rows were deleted", then the case is most probably ```ServiceNotAssignedException``` and not ```ServiceNotExistsException```. Also can be caused by the fact, that service or facility with passed ID doesn't exists, but it should be checked using own select - there are probably already existing methods for checking service/facility existence by id. Please fix javadoc for this too, your original exception is missing there. Thank you.